### ISSUE TITLE
Bug: [`ts-ban-comment`] doesn't report error for a description includes with multiple code units

### ISSUE THREAD

**Issue by sosukesuzuki** [typescript-eslint/typescript-eslint#5678]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.8.3&sourceType=module&code=PQKgpgzgNglgdgFwAQAEEE8AOkDGAnGTBAWklkWACMBDOYhCYnAewFtWxEAuJAbQCIwePMzz8AuiGAAoacGCoGpAB7YcJISLxJAvBuALHcCwBDsCWO0cDmO0cBmO9IBmzZgDoaeaUA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
/*eslint @typescript-eslint/ban-ts-comment: ["error"]*/

// @ts-expect-error 👨‍👩‍👧‍👦
foo.bar
```

### ESLint Config

```javascript
// See a comment in Repro Code

module.exports = {
  "rules": {}
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

Report an error because description length is actually 1.

### Actual Result

Does not report any errors.

### Additional Info

`String.prototype.length` isn't not enough for this rule. We should to consider to use third party package like https://github.com/orling/grapheme-splitter .

```
"👨‍👩‍👧‍👦".length; // 11
```

### Versions

| package | version |
| -- | -- |
| `@typescript-eslint/eslint-plugin` | `5.38.0` |
| `@typescript-eslint/parser` | `5.38.0` |
| `TypeScript` | `4.8.3` |
| `ESLint` | `8.15.0` |
| `node` | `web` |

**Comment by sosukesuzuki** [comment]
If this will be triaged as a bug, I can work on this.

**Comment by JoshuaKGoldberg** [comment]
Haha, I love this bug. Great. Accepting PRs to fix.

But, I don't think it's common or important enough to justify adding a new package dependency for all users to install. Can we get by with just a regular expression?

Rough examples:

```js
"👨‍👩‍👧‍👦".split(/\p{Emoji_Presentation}/ug).join("")
// ""

"before 👨‍👩‍👧‍👦 after".split(/\p{Emoji_Presentation}/ug).join("")
// "before  after"
```

There's got to be a way to get a string like `"before ! after"` (some arbitrary single length character replacing the emoji) but I don't have the time to find it just now. 😄 

https://www.regextester.com/106421
https://stackoverflow.com/questions/43242440/javascript-regular-expression-for-unicode-emoji
https://stackoverflow.com/questions/18862256/how-to-detect-emoji-using-javascript

**Comment by sosukesuzuki** [comment]
> But, I don't think it's common or important enough to justify adding a new package dependency for all users to install. Can we get by with just a regular expression?

Unfortunately, it is difficult to obtain the exact number of characters using only regular expressions...
(FYI [ESLint already uses](https://github.com/eslint/eslint/blob/main/package.json#L80) an npm package called [grapheme-splitter](https://github.com/orling/grapheme-splitter))

Not possible right now, but after Node.js v14 hits EoL and typescript-eslint only supports Node.js 16 or later, we can use [`Intl.Segmenter`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Segmenter).

**Comment by JoshuaKGoldberg** [comment]
> (FYI [ESLint already uses](https://github.com/eslint/eslint/blob/main/package.json#L80) an npm package called [grapheme-splitter](https://github.com/orling/grapheme-splitter))

Aha, perfect! If ESLint already depends on an npm package then we can take a dependency on the same version of that package. Our users won't be downloading anything extra on our account. Great spot.

**Comment by maurer2** [comment]
Hello, I believe one could also use https://github.com/surferseo/intl-segmenter-polyfill#node until support for Node 14 is dropped and then move to the native `Intl.Segmenter` as already proposed above. The polyfill adds roughly 1.3mb to the bundle size.

**Comment by JoshuaKGoldberg** [comment]
> The polyfill adds roughly 1.3mb to the bundle size.

> 1.3mb

😬
